zeppelin-server/src/main/java/org/apache/zeppelin/socket/NotebookServer.java

```diff
@@ -263,19 +263,17 @@ public void onMessage(NotebookSocket conn, String msg) {
       }
 
       TicketContainer.Entry ticketEntry = TicketContainer.instance.getTicketEntry(receivedMessage.principal);
-      if (ticketEntry != null &&
-              (!ticketEntry.getTicket().equals(receivedMessage.ticket))) {
+      if (ticketEntry == null || StringUtils.isEmpty(ticketEntry.getTicket())) {
+        LOG.debug("{} message: invalid ticket {}", receivedMessage.op, receivedMessage.ticket);
+        return;
+      } else if (!ticketEntry.getTicket().equals(receivedMessage.ticket)) {
         /* not to pollute logs, log instead of exception */
         if (StringUtils.isEmpty(receivedMessage.ticket)) {
-          LOG.debug("{} message: invalid ticket {} != {}", receivedMessage.op,
-                  receivedMessage.ticket, ticketEntry.getTicket());
-        } else {
-          if (!receivedMessage.op.equals(OP.PING)) {
-            conn.send(serializeMessage(new Message(OP.SESSION_LOGOUT).put("info",
-                    "Your ticket is invalid possibly due to server restart. "
-                            + "Please login again.")));
-          }
+          LOG.debug("{} message: invalid ticket {} != {}", receivedMessage.op, receivedMessage.ticket, ticketEntry.getTicket());
```

Comment:
`receivedMessage.ticket` is empty or null at this time. I think that this message makes no sense.

Reply:
Yeah, i'm with you on that. But it appears, as shown in the screenshot, so we log it and fast return, other than execute downstram logic, and then resulting in NPE.

Reply:
Your screenshot shows a message that is logged at [line 466](https://github.com/apache/zeppelin/blob/eb0f8ab8c07c83f054a9de23aaa4edf9239e585c/zeppelin-server/src/main/java/org/apache/zeppelin/socket/NotebookServer.java#L466).
The debug log level is turned off by default. I think we should remove the first if and always log the line in the debug log level.

Reply:
I don't quite understand what you mean, but i think a invalid ticket is no sence, we shoud log it in debug log level and fast return.

Reply:
I mean this:
```
} else if (!ticketEntry.getTicket().equals(receivedMessage.ticket)) {
  LOG.debug("{} message: invalid ticket {} != {}", receivedMessage.op, receivedMessage.ticket, ticketEntry.getTicket());
  if (!receivedMessage.op.equals(OP.PING)) {
    conn.send(serializeMessage(new Message(OP.SESSION_LOGOUT).put("info", "Your ticket is invalid possibly due to server restart. Please login again.")));
  }
  return;
}

Reply:
This is an existing code fragment, and it means send logout messages to client when invalid ticket, i think we can keep it.

Reply:
good proposal, let me update it, thx.